### DISCUSSION TITLE
Sharing Same Tests Across Files

### DISCUSSION THREAD

**Discussion by SDIjeremy** [pestphp/pest#650] [Q&A]
Maybe this is weird, but I've created Test files that all have the same dynamic tests, the only difference is the imported `Seeder::class` and what's in the `$assertions`.

I can't figure out how I can have all my dynamic Tests in just one location, instead of being repeated into all 20+ test cases.
It's becomes a real pain if I have to change one test and copy those changes to every file.

*tests/posts/p1_test.php*
```php
<?php

use Database\Seeders\Posts\Post_One as Seeder;

$assertions = [
    'post_type' => PostType::Video,
    'edits' => [
        [...],
        [...],
        [...],
    ],
];

beforeEach(function () {
    test()->post = createPost(Seeder::class);
});

///

test("Post: Type of ". Str::ucfirst($assertions['post_type']->value))
    ->expect(fn() => test()->post)
    ->post_type->toEqual($assertions['post_type']);

test("Post: Was Edited x". count($assertions['edits']))
    ->expect(fn() => closePost(Seeder::class))
    ->edits->toHaveCount(count($assertions['edits']));

...
```

**Comment by ceilidhboy** [comment]
Can this not be achieved simply by using [datasets](https://pestphp.com/docs/datasets)?

**Reply by SDIjeremy** [reply]
I'd have to try, but how would it work wit the seeders?

**Reply by ceilidhboy** [reply]
It’s hard to say from the limited code you’ve posted. Seeding is appropriate in the test set up phase, which is where you’ve put it in the beforeEach. If you need access to the data in the dataset for seeding, it may be more appropriate to seed in the test itself.

**Reply by SDIjeremy** [reply]
Each Dataset would have it's own seeder. Imagine a secondary Test file but the `Seeder::class` is different and so is the `$asertions`. I updated the above code to show that, if that makes more sense now.
